src/main/kotlin/com/cognifide/gradle/aem/common/instance/LocalInstanceManager.kt

```diff
@@ -0,0 +1,125 @@
+package com.cognifide.gradle.aem.common.instance
+
+import com.cognifide.gradle.aem.AemExtension
+import com.cognifide.gradle.aem.common.file.FileOperations
+import com.cognifide.gradle.aem.common.instance.local.BackupResolver
+import com.cognifide.gradle.aem.common.instance.local.QuickstartResolver
+import com.cognifide.gradle.aem.common.utils.Formats
+import com.cognifide.gradle.aem.common.utils.onEachApply
+import com.fasterxml.jackson.annotation.JsonIgnore
+import java.io.File
+import java.io.Serializable
+
+class LocalInstanceManager(private val aem: AemExtension) : Serializable {
+
+    /**
+     * Path in which local AEM instances will be stored.
+     */
+    var rootDir: File = aem.props.string("localInstance.root")?.let { aem.project.file(it) }
+            ?: aem.projectMain.file(".aem/instance")
+
+    /**
+     * Determines how instances will be created (from backup or quickstart built from the scratch).
+     */
+    var source = LocalInstanceSource.of(aem.props.string("localInstance.source")
+            ?: LocalInstanceSource.AUTO.name)
+
+    /**
+     * Path from which extra files for local AEM instances will be copied.
+     * Useful for overriding default startup scripts ('start.bat' or 'start.sh') or providing some files inside 'crx-quickstart'.
+     */
+    var overridesDir = File(aem.configCommonDir, LocalInstance.FILES_PATH)
+
+    /**
+     * Wildcard file name filter expression that is used to filter in which instance files properties can be injected.
+     */
+    var expandFiles: List<String> = listOf(
+            "**/start.bat",
+            "**/stop.bat",
+            "**/start",
+            "**/stop"
+    )
+
+    /**
+     * Custom properties that can be injected into instance files.
+     */
+    var expandProperties: Map<String, Any> = mapOf()
+
+    val quickstart = QuickstartResolver(aem)
+
+    fun quickstart(options: QuickstartResolver.() -> Unit) {
+        quickstart.apply(options)
+    }
+
+    val backup = BackupResolver(aem)
+
+    fun backup(options: BackupResolver.() -> Unit) {
+        backup.apply(options)
+    }
+
+    @get:JsonIgnore
+    val backupZip: File?
+        get() {
+            return when (source) {
+                LocalInstanceSource.AUTO -> backup.auto
```

Comment:
Move this logic to BackupResolver

Reply:
Nope as 'source' is not only backup related. However refactored also 😉